Hello-seqera (#413)
nextflow-io/training: docs/hello_nextflow/10_hello_seqera.md

---
title: Hello Seqera
description: Get started with Seqera Platform
---

# Part 9: Hello Seqera

So far we've been running Nextflow workflows on our local machine using the command line interface.
In this section, we'll introduce you to Seqera Platform, a powerful cloud-based platform for running, monitoring, and sharing Nextflow workflows.

!!! tip

    [Sign up](https://cloud.seqera.io/) to try Seqera for free or request a [demo](https://seqera.io/demo/) for deployments in your own on-premise or cloud environment.

> You'll learn how to use Seqera Platform within Nextflow CLI, the Seqera Platform GUI, and the API.

Reviewer: ```suggestion
You'll learn how to use Seqera Platform from the Nextflow CLI, the Seqera Platform GUI, and the API.
```

Maybe expand out CLI, GUI, API etc first time you use them? Not sure everyone will be nerdy enough to get them by default.

> You can use Seqera Platform via either the **CLI**, through the **online GUI** or through the **API**.

Reviewer: Duplicate from the above?

## 1. Use Seqera Platform to capture and monitor Nextflow jobs launched from the CLI

--8<-- "./seqera/01_run_with_cli.md"

## Online GUI

To run using the GUI, there are three main steps:

1. Create an account and login into Seqera Platform, available free of charge, at [cloud.seqera.io](https://cloud.seqera.io).
2. Create and configure a new [compute environment](https://docs.seqera.io/platform/24.1/compute-envs/overview).
3. Start [launching pipelines](https://docs.seqera.io/platform/24.1/launch/launchpad#launchpad).

> 

Reviewer: Maybe we should actually move the account creation from the note above to here, so you follow the 1,2,3 from above.

### Configuring your compute environment

Seqera Platform uses the concept of **Compute Environments** to define the execution platform where a workflow will run.

It supports the launching of workflows into a growing number of **cloud** and **on-premise** infrastructures.

![Compute environments](img/compute_env_platforms.png)

> Each compute environment must be pre-configured to enable Seqera Platform to submit tasks. You can read more on how to set up each environment using the links below.

Reviewer: Include some allusion to Forge (i.e. that we can help with that pre-configuration?)

Author: Will be addressed in later iteration.

!!! tip "The following guides describe how to configure each of these compute environments."

    * [AWS Batch](https://docs.seqera.io/platform/24.1/compute-envs/aws-batch)
    * [Azure Batch](https://docs.seqera.io/platform/24.1/compute-envs/azure-batch)
    * [Google Batch](https://docs.seqera.io/platform/24.1/compute-envs/google-cloud-batch)
    * [Google Life Sciences](https://docs.seqera.io/platform/24.1/compute-envs/google-cloud-lifesciences)
    * [HPC (LSF, Slurm, Grid Engine, Altair PBS Pro)](https://docs.seqera.io/platform/24.1/compute-envs/hpc)
    * [Amazon Kubernetes (EKS)](https://docs.seqera.io/platform/24.1/compute-envs/eks)
    * [Google Kubernetes (GKE)](https://docs.seqera.io/platform/24.1/compute-envs/gke)
    * [Hosted Kubernetes](https://docs.seqera.io/platform/24.1/compute-envs/k8s)

> 

Reviewer: We'll be teaching this somewhere a CE is already defined, which we would describe here, right?

Author: Will be included in later iteration.

### Selecting a default compute environment

If you have more than one **Compute Environment**, you can select which one will be used by default when launching a pipeline.

1. Navigate to your [compute environments](https://docs.seqera.io/platform/24.1/compute-envs/overview).
2. Choose your default environment by selecting the **Make primary** button.

**Congratulations!**

You are now ready to launch workflows with your primary compute environment.

### Launchpad

Launchpad makes it easy for any workspace user to launch a pre-configured pipeline.

![Launchpad](img/overview_launch.png)

A pipeline is a repository containing a Nextflow workflow, a compute environment and workflow parameters.

### Pipeline Parameters Form

Launchpad automatically detects the presence of a `nextflow_schema.json` in the root of the repository and dynamically creates a form where users can easily update the parameters.

!!! info

>     The parameter forms view will appear if the pipeline has a Nextflow schema file for the parameters. Please refer to the [Nextflow Schema guide](https://docs.seqera.io/platform/24.1/pipeline-schema/overview) to learn more about the schema file use-cases and how to create them.

Reviewer: ```suggestion
    The parameter forms view will only appear if the pipeline has a Nextflow schema file for the parameters. Please refer to the [Nextflow Schema guide](https://docs.seqera.io/platform/24.1/pipeline-schema/overview) to learn more about the schema file use-cases and how to create them.
```

Author: Will include in later iteration.

This makes it trivial for users without any expertise in Nextflow to enter their workflow parameters and launch.

![Pipeline parameters](img/launch_rnaseq_nextflow_schema.png)

### Adding a new pipeline

> Adding a pipeline to the pre-saved workspace launchpad is detailed in full on the [Seqera webpage docs](https://docs.seqera.io/platform/24.1/launch/launchpad#add-new-pipeline).

Reviewer: ```suggestion
Adding a pipeline to the pre-saved workspace launchpad is detailed in full on the [Seqera webpage docs](https://docs.seqera.io/platform/24.1/getting-started/quickstart-demo/add-pipelines).
```

The existing link doesn't land you anywhere useful

> In brief, these are the steps you need to follow to set up a pipeline.

Reviewer: The below steps don't make complete sense to me- are you combining pipeline creation in Launchpad, and Launching? Needs a clear separation I think.
